### DISCUSSION TITLE
(RESOLVED) device not found (indi)

### DISCUSSION THREAD

**Discussion by dcecchino** [XEphem/XEphem#41] [General]
I have a ioptron cem25p, in xephem, it is able to connect to the indi server, but when I try to select goto, it says device telescope not found. is there a config file I am missing?

**Comment by dcecchino** [comment]
[inditrace.txt](https://github.com/XEphem/XEphem/files/7535565/inditrace.txt)

inditrace log

**Comment by dcecchino** [comment]
![Screenshot from 2021-11-14 22-10-23](https://user-images.githubusercontent.com/17807052/141716445-2d4742fd-e1d3-4208-872c-812a8be8311e.png)

**Reply by dcecchino** [reply]
![Screenshot from 2021-11-14 22-21-19](https://user-images.githubusercontent.com/17807052/141717272-5b1120eb-b200-4440-a40f-8b17ccaa133f.png)

same issue in 4.1.0

**Reply by rmathar** [reply]
I suspect that is a mismatch between the device names from the INDI client which uses names like "GPSD" and "iEQ", whereas the XEphem driver uses the devices "Mount" and "Telescope". See the XEPhem database default in the file GUI/xephem/fallbacks.c in the source code. 

What I would do: to copy the Xephem*INDIConfig* X11 properties of that default into a local X11 configuration directory, likely ~/.xpehem/XEphem, and to replace Mount -> GPSD and probably Telescope -> iEQ with an editor, and to restart xephem.

**Reply by dcecchino** [reply]
Thank you, I tried that, but xephem Alert,   INDI   iEQ.SetRADec2k: number not found

**Reply by dcecchino** [reply]
Okay it was EQUATORIAL_EOD_COORD, RA/DEC doesn't give me an error anymore but now it seems that the sending edb parameter I am not getting right? Any thoughts on what that could be?

**Comment by rmathar** [comment]
From the ``inditrace.txt`` posted above one would guess that ``SetRADec2K.Dec`` must be ``TARGET_EOD_COORD.DEC`` or ``EQUATORIAL_EOD_COORD.DEC``  in the X11 properties. But since the iOptron is a Goto-compatible device, the actual command might be hidden in some GOTO command channel..(?).
The dictionary in use does apparently not stem from the vendor but from the https://indilib.org (source code in https://github.com/indilib/indi/releases).

**Reply by dcecchino** [reply]
So everything looks to be working with the following, though I get a text not found on EBD , not sure what's about, but the telescope is controllable. Enough for me to say I'm good for now. Thanks for your help! 
```
XEphem*INDIConfig*EDBValue.value:            iOptron CEM25.TARGET_EOD_COORD.edb
XEphem*INDIConfig*GotoDecValue.value:        iOptron CEM25.EQUATORIAL_EOD_COORD.
DEC
XEphem*INDIConfig*GotoRAValue.value:         iOptron CEM25.EQUATORIAL_EOD_COORD.
RA
XEphem*INDIConfig*JDValue.value:             GPSD.Date.JD
XEphem*INDIConfig*LatitudeValue.value:       GPSD.GEOGRAPHIC_COORD.LAT
XEphem*INDIConfig*LongitudeValue.value:      GPSD.GEOGRAPHIC_COORD.LONG
XEphem*INDIConfig*POSIXTMValue.value:        GPSD.POSIX
XEphem*INDIConfig*SkyDecValue.value:         iOptron CEM25.Pointing.DEC
XEphem*INDIConfig*SkyRAValue.value:          iOptron CEM25.Pointing.RA
```

**Reply by rmathar** [reply]
Because the acronym EDB seems to mean "ephemeris data base" and that is a very specific name used only inside XEphem (in particular as a suffix in file names of the various catalogs), it's unlikely that any telescope INDI server understands or maintains such an entry. That would require on the telescope side a parser for that comma-separated line format. So putting this X11 property into a comment (inserting an exclamation or hash mark at the very beginning of the line in the configuration file) ought to get rid of that error message. [Maybe EDB alternatively means "exchange data base"... I cannot tell...]

**Reply by dcecchino** [reply]
I'll give putting the exclamation or hash a try tonight, thanks for the help again!

**Reply by dcecchino** [reply]
Works perfectly! Thank you!
